JASP-Engine/JASP/R/common.R

```diff
@@ -1139,5 +1142,48 @@ saveImage <- function(plotName, format, height, width){
 	
 	updater() # create empty progressbar
 	
+	if (parallel)
+		return(structure(list(updater=updater), class="JASP-progressbar"))
+	
 	return(updater)
 }
+
+.updateParallelProgressbar <- function(progressbar, cluster, results=NULL, complete=FALSE) {
+	
+	if (! inherits(progressbar, "JASP-progressbar"))
+		stop("Object provided in 'progressbar' is not of class JASP progressbar")
+	
+	if (! inherits(cluster, "cluster"))
+		stop("Object provided in 'cluster' is not of class cluster")
+	
+	response <- progressbar$updater(results, complete)
+	
+	if (! .shouldContinue(response)) {
+		snow::stopCluster(cluster)
```

Comment:
Please add `snow` to list of dependencies if it's not present already

Reply:
This will be done automatically when requiredPackages.R is run.